src/model/entities/rundown.ts

```diff
@@ -139,11 +145,16 @@ export class Rundown extends BasicRundown {
         throw exception
       }
       this.unmarkNextSegment()
-      const segment: Segment = this.findNextSegment()
-      // TODO: Handle that we might be on the last Segment
-      if (segment) {
+      // TODO: Refactor to use less nesting
+      try {
+        const segment: Segment = this.findNextValidSegment()
         this.nextCursor = this.createCursor(this.nextCursor, { segment, part: segment.findFirstPart(), owner })
         this.markNextSegment()
+      } catch (err) {
+        if ((err as Exception).errorCode !== ErrorCode.LAST_SEGMENT_IN_RUNDOWN) {
+          throw err
+        }
+        // TODO: Notify about last Segment.
```

Comment:
Technically yes, but if we don't catch the exception here we are not able to Take the last Segment.
But in general we need better handling around the last Segment, because with this implementation we are not able to leave the Last Segment again...

I'm going to write a bug task on that.